### PR TITLE
[Typing][A-17] Add type annotations for `paddle/nn/layer/conv.py`

### PR DESCRIPTION
**Pull request by liyongchao911**
### PR Category
User Experience 

### PR Types
Improvements

### Description
类型标注：

- paddle/nn/layer/conv.py

### Related links
 
- https://github.com/PaddlePaddle/Paddle/issues/65008

@SigureMo @megemini

### CODE REVIEW

File: python/paddle/nn/layer/distance.py
Comment (SigureMo) [inline python/paddle/nn/layer/distance.py:1]: 这是之前 PR 的？merge 一下 develop

Comment (liyongchao911) [reply python/paddle/nn/layer/distance.py:1]: 我删掉重新提交

File: python/paddle/nn/layer/conv.py [hunk 1]
```diff
@@ -49,21 +55,21 @@ def _reverse_repeat_list(t, n):
...
+        dims: int,
+        stride: int | tuple | list = 1,
+        padding: int | tuple | list = 0,
+        padding_mode: str = 'zeros',
+        output_padding: int | tuple | list = 0,
+        dilation: int | tuple | list = 1,
+        groups: int = 1,
+        weight_attr: ParamAttr | None = None,
+        bias_attr: ParamAttr | bool | None = None,
+        data_format: str = "NCHW",
```
Comment (SigureMo) [inline python/paddle/nn/layer/conv.py:71]: 标注的太简单了

Comment (liyongchao911) [reply python/paddle/nn/layer/conv.py:71]: 比如说？

Comment (SigureMo) [reply python/paddle/nn/layer/conv.py:71]: https://github.com/cattidea/paddlepaddle-stubs/blob/e19bc56e035e56d5d6490c95d94570eb8eeb861f/paddle-stubs/nn/layer/conv.pyi#L44-L60

建议多看看其他人的 PR

Comment (liyongchao911) [reply python/paddle/nn/layer/conv.py:71]: OK， Thanks！

**Pull request closed without merging**